tests/unit/Core/Content/Product/SalesChannel/Listing/Processor/PagingListingProcessorTest.php

```diff
@@ -92,22 +108,46 @@ public static function provideTestPrepare(): \Generator
             'page' => 1,
             'limit' => 24,
         ];
+
+        yield 'Empty criteria, request with limit given' => [
+            'criteria' => new Criteria(),
+            'request' => new Request(['p' => 1, 'limit' => 200]),
+            'page' => 1,
+            'limit' => 200,
```

Comment:
this should be 100, right? as there is nothing given, the default should be taken